### DISCUSSION TITLE
Need Help Understanding Instructions

### DISCUSSION THREAD

**Discussion by ShadowVR2** [Quitch/New-GW-Cards#3] [Q&A]
Trying to make sense of some of the steps involved with creating new loadouts, but I'm still confused by certain instructions that lack elaboration (probably because it would make sense to people who do this as a hobby or for a living, idk because I'm not at that stage myself yet).

1.I'm new to using Coherent UI Debugger. What's meant to happen when I press the "Go" button? All I have is a URL and a blank white page. [URL: http://127.0.0.1:9999]
2.I don't know how to "Change to the Console tab". What do I need to do to access it?
3.I'm not sure to make my loadouts actually show up. The most I had were two blank bars that I could select in the loadout list. I feel like I'm missing an instruction or misunderstanding an instruction.

Would appreciate any help with making sense of the stuff here so I can make cards. Mostly for personal use, but I wouldn't mind sharing some loadouts (when I get it working).

**Comment by Quitch** [comment]
1. What you're describing is what you see on loading the debugger. Below is a screenshot of the debugger while PA is on the main menu and the go button was pressed
![image](https://user-images.githubusercontent.com/5182715/193537287-88235469-72e1-417c-b75a-57997a3e48c0.png)

2. You would need to have clicked Start Page to get this tab.

3. I'd need a lot more detail, since there's going to be an error in your implementation somewhere. Two blank lines would sound like you haven't set up the card files properly, or the game can't find them. The console would give more info on the error encountered.

**Reply by ShadowVR2** [reply]
![Screenshot 2022-10-03 183251](https://user-images.githubusercontent.com/97499594/193709580-f6df9800-3d21-4ab6-8c94-ed553ca44d13.png)
This is all the debugger is showing me. Why does nothing show up for me?
Does it need time to load, should it instantly appear upon pressing Go, do I need PA to be in windowed mode?

**Comment by Quitch** [comment]
I forgot that a change was made to PA where to allow connections from the debugger, you will need to add the following launch option to the game `--coherent_port=9999`

I'll add that to the instructions somewhere. I think that should fix your issue.

**Comment by ShadowVR2** [comment]
![Screenshot 2022-10-04 115909](https://user-images.githubusercontent.com/97499594/193892228-5161232d-109a-4f36-b2f7-adc7078bbc2a.png)
![Screenshot 2022-10-04 115339](https://user-images.githubusercontent.com/97499594/193892252-79756b18-85a1-459e-8ebb-dcba18bc68c4.png)
Got unexpected token for ")", but can't seem to see where it would crop up.
(The censored part was due to including identity in it. Will change later, but for for purposes of this trust me when I say it is the exact same for both lines.)

**Reply by Quitch** [reply]
You're getting the bracket error because you commented out lines 9 through 11, so there's nothing inside the push.

**Reply by ShadowVR2** [reply]
Got that, thanks. Also it gave the same error again, but turns out that having a comma on a singular item screws with it (weird that it calls out the righthand bracket and not the comma). Now it show slots for both loadouts that aren't one-third the size of the rest, however I'm getting a different error.
![Screenshot 2022-10-04 142135](https://user-images.githubusercontent.com/97499594/193919761-e925bbdd-7029-42f3-8d8b-852e0c5eb9ba.png)
![Screenshot 2022-10-04 142334](https://user-images.githubusercontent.com/97499594/193919796-8c0c7793-ab03-437b-9c7b-f6a97a92e236.png)
(Just gonna focus on this one, since they probably have the same problem, and also combat com code is _long_.)

**Reply by Quitch** [reply]
You'll need to expand the errors and track down the root cause.

**Reply by ShadowVR2** [reply]
Oh that's how you do it. Didn't notice that.
![Screenshot 2022-10-04 145736](https://user-images.githubusercontent.com/97499594/193926852-aec9b5c7-364f-4330-90e8-98cc38c756f1.png)

**Reply by Quitch** [reply]
My guess would be you've put the cards under `/ui/mods/...` but you need to put them under `/ui/main/game/galactic_war/cards`

**Reply by ShadowVR2** [reply]
But I am though?
![Screenshot 2022-10-04 152913](https://user-images.githubusercontent.com/97499594/193933813-7631ba9a-619f-468e-bf27-c5df34f61ab0.png)
Unless you mean somewhere else?

**Reply by Quitch** [reply]
That path is wrong, it doesn't match the path I provided.

**Reply by ShadowVR2** [reply]
...Just noticed my mistake.

**Reply by ShadowVR2** [reply]
Is this better?
![Screenshot 2022-10-04 161200](https://user-images.githubusercontent.com/97499594/193939690-6fd989ef-222d-4eb2-952f-23c39ac91625.png)

**Reply by Quitch** [reply]
No. All you needed was the ui folder you were missing in your previous path.

**Reply by ShadowVR2** [reply]
![Screenshot 2022-10-05 100044](https://user-images.githubusercontent.com/97499594/194107468-0fac2d0e-5d31-425a-8c91-bee2d4d6bd7e.png)

**Reply by ShadowVR2** [reply]
![Screenshot 2022-10-05 180535](https://user-images.githubusercontent.com/97499594/194185428-14fda6ff-6c13-45f4-9117-08ba3e003947.png)
![Screenshot 2022-10-05 180610](https://user-images.githubusercontent.com/97499594/194185436-90b84be4-565e-46dd-8434-d160633ca218.png)

**Reply by Quitch** [reply]
I'm afraid there's a lot of things that could cause that. You're going to need to step through everything, double and triple check your work. You may wish to make use of the #modding room on the [PA Discord](https://discord.gg/pa).

**Reply by ShadowVR2** [reply]
Thank you for the help thus far.

**Reply by ShadowVR2** [reply]
I'm back, and I've been sent back to you about some errors that cropped up.
![Screenshot 2022-10-19 133727](https://user-images.githubusercontent.com/97499594/196788016-44b9a002-f008-47c8-b4c8-715f624dcc58.png)
![Screenshot 2022-10-19 133903](https://user-images.githubusercontent.com/97499594/196788044-908bee65-d43d-459b-bdff-5b7a7aed0a07.png)
These came up when using my test start (gwal_test_start in the code) when trying to start a fight, which have similarly came up in other custom starts. A notable example being Space Excavation Alt (originally my only successful attempt at getting reworked loadouts to function), which suddenly is being stuck with the same slew of errors. I'm not sure what I did to throw everything into chaos.
[AmalgamLoadouts.zip](https://github.com/Quitch/New-GW-Cards/files/9823783/AmalgamLoadouts.zip)

**Reply by Quitch** [reply]
All the 404s mean something is being referenced which doesn't exist i.e. your path is wrong.

**Reply by ShadowVR2** [reply]
That's what I don't understand. I had to have changed something, because even if I do the bare minimum to port pre-existing loadouts to my mod I still get thrown this error. What is the deal with PATH 1, PATH 2, etc?

**Reply by Quitch** [reply]
They're place holders for you to replace per the comments above them.

**Reply by ShadowVR2** [reply]
But what file?

**Reply by Quitch** [reply]
What file what?

**Reply by ShadowVR2** [reply]
Where those placeholders are found. I don't know where to find them. Also I don't know how they weren't a problem until now.

**Reply by ShadowVR2** [reply]
Another thing I'm noticing that I didn't cover yet was a message when in the galactic war menu.
![Screenshot 2022-10-20 225650](https://user-images.githubusercontent.com/97499594/197115490-b58a3243-abc7-456f-8a74-d3c216e010ee.png)
Don't know if this helps Identify my current problem.

**Reply by Quitch** [reply]
For every 404 you need to find those paths and correct them because a 404 means there's nothing found at the current path. The comments above every path area tell you what you need to be entering, depending on your card.

**Reply by ShadowVR2** [reply]
I don't know where within my mod this path is used. This seems to be a path that's only used by the base game.

**Reply by Quitch** [reply]
I'm afraid I'm not going to be around for a while, so you might want to ask on Discord. There's quite a few areas that need fixing.
